### PR TITLE
Types, PEP517, GitHub Actions, etc.

### PR DESCRIPTION
**Pull request by akx**
Pretty much just the usual plumbing to get type annotations in so downstream projects can be more secure about them.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  pull_request:
+    branches:
+      - master
+jobs:
+  Test:
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        python-version:
+          - "3.6"
```
Comment (tokkoro) [inline .github/workflows/ci.yml:15]: Should the minimum version be 3.7 as the 3.6 is at the end of its life cycle?

File: tox.ini [hunk 2]
```diff
@@ -1,5 +1,6 @@
 [tox]
-envlist = {py27,py34,py35}
+isolated_build = true
+envlist = {py36,py310}
```
Comment (tokkoro) [inline tox.ini:3]: Same python 3.6 question here.

File: setup.cfg [hunk 3]
```diff
@@ -1,5 +1,21 @@
...
+name = koodaus
+version = attr:koodaus.__version__
+description = Encoding/decoding library for Python
+author = Valohai
+author_email = info@valohai.com
+url = https://github.com/valohai/koodaus
+license = MIT
+
+[options]
+python_requires = >=3.6
```
Comment (tokkoro) [inline setup.cfg:12]: And here

**Review by tokkoro — APPROVED** [review]
If python 3.6 is necessary, then good to go :checkered_flag:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (akx) [reply .github/workflows/ci.yml:15]: Unfortunately not – we still need to support 3.6 for certain environments.

**Pull request merged**